base/workerpool.jl

```diff
@@ -11,59 +11,77 @@ abstract AbstractWorkerPool
 # `isready` - return false if a `take!` on the pool would block, else true
 #
 # The default implementations of the above (on a AbstractWorkerPool) require fields
-#    channel::RemoteChannel{Channel{Int}}
+#    channel::Channel{Int}
 #    workers::Set{Int}
 #
 
 type WorkerPool <: AbstractWorkerPool
-    channel::RemoteChannel{Channel{Int}}
+    channel::Channel{Int}
     workers::Set{Int}
+    ref::RemoteChannel
 
-    # Create a shared queue of available workers
-    WorkerPool() = new(RemoteChannel(()->Channel{Int}(typemax(Int))), Set{Int}())
-end
-
-
-"""
-    WorkerPool(workers)
+    function WorkerPool()
+        wp = WorkerPool(Channel{Int}(typemax(Int)), RemoteChannel())
+        put!(wp.ref, wp)
+        wp
+    end
 
-Create a WorkerPool from a vector of worker ids.
-"""
-function WorkerPool(workers::Vector{Int})
-    pool = WorkerPool()
+    """
```

Comment:
This needs a `@doc` I believe, see the warning from `make docs` - and there was another change that needed a doc update for "used by ``remote(f)`` and ``pmap`` (by default)"